platform/openide.util/test/unit/src/org/openide/util/PairTest.java

```diff
@@ -19,39 +19,90 @@
+
+        assertTrue(pair11a.equals(pair11b));
+        assertEquals(pair11a.hashCode(), pair11b.hashCode());
+        assertEquals(new Integer(1), pair11b.first());
+        assertEquals(new Integer(1), pair11b.second());
+
+        assertTrue(pair1Null.equals(pair1Null));
+        assertEquals(pair1Null.hashCode(), pair1Null.hashCode());
+        assertEquals(new Integer(1), pair1Null.first());
+        assertEquals(null, pair1Null.second());
+
+        assertTrue(pairNul1.equals(pairNul1));
+        assertEquals(pairNul1.hashCode(), pairNul1.hashCode());
+        assertEquals(null, pairNul1.first());
+        assertEquals(new Integer(1), pairNul1.second());
+
+        assertTrue(pairNullNull.equals(pairNullNull));
+        assertEquals(pairNullNull.hashCode(), pairNullNull.hashCode());
+        assertEquals(null, pairNullNull.first());
+        assertEquals(null, pairNullNull.second());
+    }
+
+    //-------------------------------------------------------------------------- 
+    public void test_pairInequality() {
+
+        Pair<Integer, Integer> pair11a = Pair.of(1, 1);
+        Pair<Integer, Integer> pair12 = Pair.of(1, 2);
+        Pair<Integer, Integer> pair21 = Pair.of(2, 1);
+        Pair<Integer, Integer> pairNul1 = Pair.of(null, 1);
+        Pair<Integer, Integer> pair1Null = Pair.of(1, null);
+        Pair<Integer, Integer> pairNullNull = Pair.of(null, null);
+        
+        assertFalse(pair11a.equals(pair12));
+        assertFalse(pair11a.equals(pair21));
+        assertFalse(pair11a.equals(pairNul1));
+        assertFalse(pair11a.equals(pair1Null));
+        assertFalse(pair11a.equals(pairNullNull));
+        assertFalse(pairNul1.equals(pair11a));
+        assertFalse(pair1Null.equals(pair11a));
+        assertFalse(pairNullNull.equals(pair11a));
 
-    public void testPairs() {
-        final Pair<Integer,Integer> p1a = Pair.of(1, 1);
-        final Pair<Integer,Integer> p1b = Pair.of(1, 1);
-        final Pair<Integer,Integer> p2 = Pair.of(1, 2);
-        final Pair<Integer,Integer> p3 = Pair.of(2, 1);
-        final Pair<Integer,Integer> p4 = Pair.of(null, 1);
-        final Pair<Integer,Integer> p5 = Pair.of(1, null);
-        final Pair<Integer,Integer> p6 = Pair.of(null, null);
-        assertTrue(p1a.equals(p1a));
-        assertTrue(p1a.equals(p1b));
-        assertFalse(p1a.equals(p2));
-        assertFalse(p1a.equals(p3));
-        assertFalse(p1a.equals(p4));
-        assertFalse(p1a.equals(p5));
-        assertFalse(p1a.equals(p6));
-        assertEquals(p1a.hashCode(), p1b.hashCode());
-        assertEquals(p4.hashCode(), p4.hashCode());
-        assertEquals(p5.hashCode(), p5.hashCode());
-        assertEquals(p6.hashCode(), p6.hashCode());
-        assertFalse(p4.hashCode() == p5.hashCode());
+        assertFalse(pair11a.equals(null));
+        assertFalse(pair11a.equals(""));
     }
 
+    //--------------------------------------------------------------------------
+    public void test_pair_toString() {
```

Comment:
Please use camelCase and not snakeCase

Reply:
fixed in https://github.com/apache/netbeans/pull/5115/commits/68bc78dc9355a667f7d8208c9e0f39bf53c18013